docs/ja/docs/tutorial/security/oauth2-jwt.md

```diff
@@ -0,0 +1,266 @@
+# パスワード（およびハッシュ化）によるOAuth2、JWTトークンによるBearer
+
+これでセキュリティの流れが全てわかったので、<abbr title="JSON Web Tokens">JWT</abbr>トークンと安全なパスワードのハッシュ化を使用して、実際にアプリケーションを安全にしてみましょう。
+
+このコードは、アプリケーションで実際に使用したり、パスワードハッシュをデータベースに保存するといった用途に利用できます。
+
+本章では、前章の続きから始めて、コードをアップデートしていきます。
+
+## JWT について
+
+JWTとは「JSON Web Tokens」の略称です。
+
+JSONオブジェクトをスペースのない長く密集した文字列で表現したトークンの仕様です。例えば次のようになります：
+
+```
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
+```
+
+これらは暗号化されていないので、誰でもコンテンツから情報を復元できてしまいます。
+
+しかし、トークンは署名されているため、あなたが発行したトークンを受け取った人は、あなたが実際に発行したということを検証できます。
+
+例えば、1週間の有効期限を持つトークンを作成したとします。ユーザーが翌日そのトークンを持って戻ってきたとき、そのユーザーはまだシステムにログインしていることがわかります。
+
+1週間後、トークンが期限切れとなるとどうなるでしょうか？ユーザーは認可されず、新しいトークンを得るために再びサインインしなければなりません。また、ユーザー（または第三者）がトークンを修正して有効期限を変更しようとした場合、署名が一致しないため、トークンの修正を検知できます。
+
+JWT トークンを使って遊んでみたいという方は、<a href="https://jwt.io/" class="external-link" target="_blank">https://jwt.io</a> をチェックしてください。
+
+## `python-jose` のインストール
+
+PythonでJWTトークンの生成と検証を行うために、`python-jose`をインストールする必要があります：
+
+<div class="termy">
+
+```console
+$ pip install python-jose[cryptography]
+
+---> 100%
+```
+
+</div>
+
+また、<a href="https://github.com/mpdavis/python-jose" class="external-link" target="_blank">Python-jose</a>だけではなく、暗号を扱うためのパッケージを追加で必要とします。
+
+ここでは、推奨されているものを使用します：<a href="https://cryptography.io/" class="external-link" target="_blank">pyca/cryptography</a>。
+
+!!! tip "豆知識"
+    このチュートリアルでは以前、<a href="https://pyjwt.readthedocs.io/" class="external-link" target="_blank">PyJWT</a>を使用していました。
+
+    しかし、Python-joseは、PyJWTのすべての機能に加えて、他のツールと統合して構築する際に必要となるいくつかの追加機能を提供しています。そのため、代わりにPython-joseを使用するように更新されました。
+
+## パスワードのハッシュ化
+
+「ハッシュ化」とは、あるコンテンツ（ここではパスワード）を、規則性のないバイト列（単なる文字列）に変換することです。
+
+特徴として、全く同じ内容（全く同じパスワード）を渡すと、全く同じ規則性のないバイト列に変換されます。
+
+しかし、規則性のないバイト列からパスワードに戻すことはできません。
+
+### パスワードのハッシュ化を使う理由
+
+データベースが盗まれても、ユーザーの平文のパスワードは盗まれず、ハッシュ値だけが盗まれます。
+
+したがって、泥棒はそのパスワードを別のシステムで使えません（多くのユーザーはどこでも同じパスワードを使用しているため、危険性があります）。
+
+## `passlib` のインストール
+
+PassLib は、パスワードのハッシュを処理するための優れたPythonパッケージです。
+
+このパッケージは、多くの安全なハッシュアルゴリズムとユーティリティをサポートします。
+
+推奨されるアルゴリズムは「Bcrypt」です。
+
+そのため、Bcryptを指定してPassLibをインストールします：
+
+<div class="termy">
+
+```console
+$ pip install passlib[bcrypt]
+
+---> 100%
+```
+
+</div>
+
+!!! tip "豆知識"
+    `passlib`を使用すると、**Django**や**Flask**のセキュリティプラグインなどで作成されたパスワードを読み取れるように設定できます。
+
+    例えば、Djangoアプリケーションからデータベース内の同じデータをFastAPIアプリケーションと共有できるだけではなく、同じデータベースを使用してDjangoアプリケーションを徐々に移行することもできます。
+
+    また、ユーザーはDjangoアプリまたは**FastAPI**アプリからも、同時にログインできるようになります。
+
+
+## パスワードのハッシュ化と検証
+
+必要なツールを `passlib`からインポートします。
+
+PassLib の「context」を作成します。これは、パスワードのハッシュと検証に使用されるものです。
+
+!!! tip "豆知識"
+    PassLibのcontext には、非推奨の古いものだけを含む様々なハッシュアルゴリズムを使用した検証機能もあります。
```

Comment:
Need to add translation of "only to allow verifying them (passwords), etc."
Actually this implies "it should not be used for hashing any more", but the original sentence does not have it, so skipping.

```suggestion
    PassLibのcontextには、検証だけが許された非推奨の古いハッシュアルゴリズムを含む、様々なハッシュアルゴリズムを使用した検証機能もあります。
```

Reply:
thanks!